src/AddressRegistry/StreetName/StreetName.cs

```diff
@@ -236,9 +236,15 @@ public void CorrectAddressApproval(AddressPersistentLocalId addressPersistentLoc
 
         public void CorrectAddressRetirement(AddressPersistentLocalId addressPersistentLocalId)
         {
-            StreetNameAddresses
-                .GetNotRemovedByPersistentLocalId(addressPersistentLocalId)
-                .CorrectRetirement();
+            var addressToCorrect = StreetNameAddresses
+                .GetNotRemovedByPersistentLocalId(addressPersistentLocalId);
+
+            GuardAddressIsUnique(
```

Comment:
We're going to have to pass the GuardAddressIsUnique method as an action to CorrectRetirement in order to execute the address IsRemoved and Status checks prior to it. See RejectAddress